### ISSUE TITLE
Idea: Keyboard shortcuts

### ISSUE THREAD

**Issue by ttofis** [synac-chat/client-gtk#3]
- [ ] Tab for example to switch channels
- [ ] Shift tab to reverse switch
- [ ] Escape to leave the chat box
- [ ] Enter to enter the chat box
- [ ] Ctrl-Tab to switch servers
- [ ] Ctrl-Shift-Tab to reverse switch servers
- [x] Up arrow to edit last sent message of yours (like discord)

**Comment by jD91mZM2** [comment]
Ctrl+[0-9] to switch servers, maybe? Ctrl+1 is 1, Ctrl+2 is 2, Ctrl+0 is 10?

**Comment by ttofis** [comment]
You will only have 10 servers? That's bad thinking

Also it should be pretty difficult to press it by accident

**Comment by jD91mZM2** [comment]
No, but if you have more than 2 it gets tedious tabbing through them all

**Comment by ttofis** [comment]
ehm why? You just hold ctrl and press tab. Also give like a 2 second time period before loading in the first channel.

**Comment by jD91mZM2** [comment]
Or I just make it select the channel instead of holding it down so you have to press enter when you find the right one.

**Comment by ttofis** [comment]
And escape then can also be for entering the chat box.

**Comment by jD91mZM2** [comment]
Okay, I found something interesting. GTK makes it really easy to make a button with an ALT+shortcut. Like I can make Alt+A trigger `Add...`. It even shows an underscore on the button when you hold down ALT by itself!

**Comment by ttofis** [comment]
That's pretty cool! Dunno if we could use it on servers tho, since a) putting a character next to them is, not a great idea b) they might have similar characters (if you consider getting them from the middle of the name). One thing is search, but if gtk doesn't have it built in, don't bother.

**Comment by jD91mZM2** [comment]
Tab and Shit+Tab are already a thing btw. They just aren't specific to servers/channels, but rather go through all of them. I'd recommend keeping it that way, and if necessary using some other shortcut to toggle channels specifically.
